### DISCUSSION TITLE
Why can't I export incrementally?

### DISCUSSION THREAD

**Discussion by earthmoon** [retorquere/zotero-better-bibtex#2646] [Ideas]
Hello,

When the library contains more than 3000 references, exporting it when changed will cause the Zotero to stuck. But choosing **exported when idle** sometimes not meet our requirement. As we known, the DropBox using incrementally sync, about which I don't know whether it is difficult. However, if it isn't, this tactic can be used here for exporting when change to avoid stucking.

Thanks.

**Comment by retorquere** [comment]
With the cache on, BBT effectively does incremental export

**Reply by earthmoon** [reply]
So, why with the cache on, Zotero will struck when **exported by changed**.

**Reply by retorquere** [reply]
Probably because you are getting an out of memory error? That's what we established earlier.

**Reply by earthmoon** [reply]
Okay. Thanks.

**Reply by earthmoon** [reply]
Do you need the bug ID, about which the automatically export will cause stuck of Zotero. From that you can tell whether the BetterBibTeX used the cached contents or always export a totally new one?

**Reply by retorquere** [reply]
If you want to be sure, then yes.

**Reply by earthmoon** [reply]
Of course, although I don't know too much about those questions, but I still want to know more about it.  The debug ID is `RB29YDY9-apse/6.7.116.2641.5050-6`. Thanks.

**Reply by retorquere** [reply]
There's no out of memory error (yet), and I see exports succeeding.

**Reply by earthmoon** [reply]
But why any changes in Zotero will cause stuck? If the number of references is smaller than 2000, no stuck.  May you tell me if BetterBibTeX used the cached files? If BetterBibTeX always export a new one, they will be stuck.

**Reply by retorquere** [reply]
without a debug log, I'd be guessing.

**Reply by earthmoon** [reply]
In the previous one `RB29YDY9-apse/6.7.116.2641.5050-6`, the Zotero is stuck for several seconds to let BetterBibTex automatically export. Thanks.

**Reply by retorquere** [reply]
This is starting to look like problem diagnosis rather than a discussion; please open an issue if we need to get deeper into this, my support tooling is built around issues.

**Reply by earthmoon** [reply]
It seems the newest version of BetterBibTeX have deal with the struck caused by automatically exporting. Now, there is no problem. If I see this problem in the future, I will post it here. Thanks.

**Reply by retorquere** [reply]
If you see it again, please open an issue instead. What we're doing here is not a discussion, but problem diagnosis.
